modular_ss220/stack_craft_tgui/code/stack.dm

```diff
@@ -0,0 +1,122 @@
+/obj/item/stack/Topic(href, href_list)
+	return
+
+/obj/item/stack/list_recipes(mob/user, recipes_sublist)
+	return
+
+/datum/stack_recipe/post_build(mob/user, obj/item/stack/S, obj/item/stack/created)
+	return
+
+/obj/item/stack/attack_self(mob/user)
+	ui_interact(user)
+
+/obj/item/stack/attack_self_tk(mob/user)
+	ui_interact(user)
+
+/obj/item/stack/attack_tk(mob/user)
+	if(user.stat || !isturf(loc))
+		return
+	// Allow remote stack splitting, because telekinetic inventory managing
+	// is really cool
+	if(src in user.tkgrabbed_objects)
+		var/obj/item/stack/F = split(user, 1)
+		F.attack_tk(user)
+		if(src && user.machine == src)
+			ui_interact(user)
+	else
+		..()
```

Comment:
```suggestion
	// Allow remote stack splitting, because telekinetic inventory managing
	// is really cool
	if(!(src in user.tkgrabbed_objects))
		..()
		return
		
	var/obj/item/stack/F = split(user, 1)
	F.attack_tk(user)
	if(src && user.machine == src)
		ui_interact(user)
```